junit-platform-launcher/src/main/java/org/junit/platform/launcher/ExcludeMethodFilter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.launcher;
+
+import static org.junit.platform.engine.FilterResult.excluded;
+import static org.junit.platform.engine.FilterResult.included;
+
+import java.util.regex.Pattern;
+
+import org.junit.platform.engine.FilterResult;
+import org.junit.platform.engine.TestDescriptor;
+
+/**
+ * {@link MethodFilter} that matches fully qualified method names against
+ * patterns in the form of regular expressions.
+ *
+ * <p>If the fully qualified name of a method matches against at least one
+ * pattern, the class will be excluded.
+ *
+ * @since 1.12
+ */
+class ExcludeMethodFilter extends AbstractMethodFilter {
+
+	ExcludeMethodFilter(String... patterns) {
+		super(patterns);
+	}
+
+	@Override
+	public FilterResult apply(TestDescriptor descriptor) {
+		String methodName = getFullyQualifiedMethodNameFromDescriptor(descriptor);
+		return findMatchingPattern(methodName) //
+				.map(pattern -> excluded(formatExclusionReason(methodName, pattern))) //
+				.orElseGet(() -> included(formatInclusionReason(methodName)));
+	}
+
+	private String formatInclusionReason(String methodName) {
+		return String.format("Method name [%s] does not match any excluded pattern: %s", methodName,
```

Comment:
```suggestion
		return String.format("Method name [%s] does not match any excluded pattern: '%s'", methodName,
```

Please fix this for `ClassNameFilter` as well.

Reply:
If I change `%s` to `'%s'` here, the returned string will look like `''methodA''` and break the tests. Because the `patternDescription` in `AbstractMethodFilter` and `AbstractClassNameFilter` is defined as:

`this.patternDescription = Arrays.stream(patterns).collect(joining("' OR '", "'", "'"));`

It already includes single quotation marks.

Should I leave it as is, or modify it and adjust the definition of `patternDescription` to:
`this.patternDescription = String.join("' OR '", patterns);`

Reply:
If I change it here, the returned string will look like ''methodA'', which would break the tests. This is because the `patternDescription` in `AbstractMethodFilter` and `AbstractClassNameFilter` is defined as:

`this.patternDescription = Arrays.stream(patterns).collect(joining("' OR '", "'", "'"));`

It already includes single quotation marks. Should I leave it as is, or modify the `patternDescription` to:

`this.patternDescription = String.join("' OR '", patterns);`

and update all usages of `patternDescription` from `%s` to `'%s'` accordingly?

Reply:
Ah, sorry, I missed that! In that case, all good, no need to change anything! :+1: